R/threshold_perf.R

```diff
@@ -89,12 +91,34 @@ threshold_perf.data.frame <- function(.data,
                                       estimate,
                                       thresholds = NULL,
                                       na_rm = TRUE,
-                                      ...) {
+                                      ...,
+                                      metrics = NULL) {
 
   if (is.null(thresholds)) {
     thresholds <- seq(0.5, 1, length = 21)
   }
 
+  if (is.null(metrics)) {
+    metrics <- two_class
+    compute_distance_metric <- TRUE
+  } else {
+    # Verify that `metrics` is an appropriate metric set
+    is_numeric_metric_set <- inherits(metrics, "numeric_metric_set")
+    is_class_prob_metric_set <- inherits(metrics, "class_prob_metric_set")
+
+    if (!is_numeric_metric_set && !is_class_prob_metric_set) {
+      rlang::abort("The `metrics` argument should be the results of [yardstick::metric_set()].")
+    }
```

Comment:
This code is borrowed from `tune::check_metrics()`. See here: https://github.com/tidymodels/tune/blob/a97576f3f3e36f362388de7e2f3ef2df8ab3a38f/R/checks.R#L305